sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/GenericTypeParameterUnused.cs

```diff
@@ -135,14 +136,15 @@ private static bool IsMethodCandidate(MethodDeclarationSyntax methodDeclaration,
                 methodSymbol.IsChangeable();
         }
 
-        private static List<string> GetUsedTypeParameters(IEnumerable<SyntaxNode> declarations,
+        private static List<string> GetUsedTypeParameters(IEnumerable<SyntaxNode> declarations, List<string> typeParameterNames,
             SyntaxNodeAnalysisContext localContext,
             Compilation compilation)
         {
             return declarations
                 .SelectMany(declaration => declaration.DescendantNodes())
                 .OfType<IdentifierNameSyntax>()
                 .Where(identifier => !(identifier.Parent is TypeParameterConstraintClauseSyntax))
+                .Where(identifier => typeParameterNames.Contains(identifier.Identifier.ValueText))
```

Comment:
given that `typeParameterNames` is used only for search, in a loop, would an `ISet` be better than `List`?